packages/calypso-e2e/src/rest-api-client.ts

```diff
@@ -0,0 +1,91 @@
+import fetch, { RequestInit } from 'node-fetch';
+
+type RequestMethods = 'post' | 'get';
+type RequestParams = {
+	method: 'post' | 'get';
+	headers: GenericDict;
+	body?: GenericDict;
+};
+type GenericDict = { [ key: string ]: string | number };
+
+interface Site {
+	ID: string;
+	name: string;
+	description: string;
+	URL: string;
+	site_owner: string;
+}
+interface MySitesResponse {
+	sites: Site[];
+}
+
+const BASE_URL = 'https://public-api.wordpress.com/rest/v1.1/';
+
+/**
+ * Client to interact with the WordPress.com REST API.
+ */
+export class RestAPIClient {
+	readonly bearerToken: string;
+
+	/**
+	 * Constructs an instance of the API client.
+	 */
+	constructor( bearerToken: string ) {
+		this.bearerToken = bearerToken;
+	}
+
+	/**
+	 * Builds and returns the request parameters.
+	 *
+	 * @param {RequestMethods} method Supported request methods.
+	 * @param {GenericDict} [body] Request body.
+	 * @returns {RequestParams} Completed request parameter.
+	 */
+	private buildRequestParams( method: RequestMethods, body?: GenericDict ): RequestParams {
+		const params: RequestParams = {
+			method: method,
+			headers: {
+				Authorization: `Bearer ${ this.bearerToken }`,
+				'Content-Type': 'application/json',
+			},
+		};
+
+		// POST requests require a body.
+		if ( method === 'post' ) {
+			if ( ! body ) {
+				throw new Error( 'Cannot build a POST request with undefined body.' );
+			}
+			params[ 'body' ] = body;
+		}
+
+		return params;
+	}
+
+	/**
+	 * Sends the request to the endpoint.
+	 *
+	 * @param {string} endpoint Endpoint to send the request.
+	 * @param {RequestParams} params Parameters for the request.
+	 */
+	private async sendRequest( endpoint: string, params: RequestParams ): Promise< any > {
+		// Trim the initial forward slash.
+		if ( endpoint.startsWith( '/' ) ) {
+			endpoint = endpoint.slice( 1 );
+		}
```

Comment:
As a heads up, if you keep using the `URL` constructor below, you don't need to do this trim -- it handles it for you! 🎉

Reply:
In my experience, this is not the case:

```typescript
await fetch( new URL('/me', 'https://public-api.wordpress.com/rest/v1.1/')
```

results in 

```
{
  error: 'not_found',
  message: 'The specified path was not found. Please visit P2gHKz-8-p2 for valid paths.'
}
```

Reply:
Interesting, so if I have it set up as follows:

```
new URL('/me', 'https://public-api.wordpress.com/rest/v1.1')
```

The resulting `URL` object looks like this:

```
URL {
  href: 'https://public-api.wordpress.com/me',
  origin: 'https://public-api.wordpress.com',
  protocol: 'https:',
  username: '',
  password: '',
  host: 'public-api.wordpress.com',
  hostname: 'public-api.wordpress.com',
  port: '',
  pathname: '/me',
  search: '',
  searchParams: URLSearchParams {},
  hash: ''
}
```

This is obviously incorrect hence the error!

If I implement the changes you suggest in https://github.com/Automattic/wp-calypso/pull/63312#discussion_r866887864 and make the request methods specify the path starting from the hostname, then it works.

Reply:
Yeah, I think based on the [MDN docs](https://developer.mozilla.org/en-US/docs/Web/API/URL/URL), they are expecting that "base" parameter to just be a host! Kind of confusing, IMHO! 🤷

Reply:
Yep, no problem - I've taken your advice to have smaller methods that build out individual parts of the request params and have made changes to the way I piece together URLs as part of that change.